### DISCUSSION TITLE
Not all values added with useFieldArray are available in getValues

### DISCUSSION THREAD

**Discussion by wondering639** [react-hook-form/react-hook-form#5159] [Q&A] [answered]
I add a new item using the `append `method from `useFieldArray`:
`append({ Name: 'K04 full description', SomeProperty: 'something' });`

Checking the values using `getValues` from `useForm` only gives me the _Name_ property, the _SomeProperty_ is missing:
`0: {Name: "K04 full description"}`

I guess it's because inside the form I only render the `Name` property, not the _SomeProperty_. If I want _SomeProperty_ to be included in the output from `getValues`, is there any other way to get it there than creating a hidden input field? Any other way to access the values provided to the `append `method?

Here's my full example:

```
import { useFieldArray } from 'react-hook-form';
import { FC } from 'react';
import { useForm } from 'react-hook-form';

interface Props {}

interface IItem {
  Name: string;
  SomeProperty: string;
}

interface FormValues {
  Items: IItem[];
}

export const ItemList: FC<Props> = () => {
  const { control, register, handleSubmit, getValues } = useForm<FormValues>(
    {}
  );

  const { fields, append } = useFieldArray<IItem>({
    control,
    name: 'Items'
  });

  const onSubmit = async (values: FormValues) => {
    console.log('submit form');
  };

  return (
    <>
      <form onSubmit={handleSubmit(onSubmit)} noValidate>
        <table>
          <tbody>
            {fields.map((item, index) => (
              <tr key={item.id}>
                <td>
                  <input
                    type="text"
                    name={`Item[${index}].Name`}
                    ref={register()}
                    defaultValue={item.Name}
                  />
                </td>
              </tr>
            ))}
          </tbody>
        </table>
        <button
          type="button"
          onClick={() => {
            append({ Name: 'K04 full description', SomeProperty: 'something' });
          }}
        >
          add more of K04
        </button>
        <button
          type="button"
          onClick={() => {
            console.log('dumping current values:');
            console.log(getValues());
          }}
        >
          get values (see console output)
        </button>
      </form>
    </>
  );
};

```

**Comment by bluebill1049** [comment, marked as answer]
share a codesandbox and provide reproduced steps.

**Reply by wondering639** [reply]
OK, find it here: https://codesandbox.io/s/reverent-dewdney-lns1l?file=/src/ItemList.tsx
Steps to reproduce:
1) Click button _add more of K04_
2) Click button _get values (see console output)_, this will give the following:
![image](https://user-images.githubusercontent.com/25722498/118022873-28a27d00-b35d-11eb-9221-ec055f52a166.png)
Note that step 2 only gives an object with the Name property, while the object was added with a _Name_ **and** _SomeProperty_ property:
```
  <button
    type="button"
    onClick={() => {
      append({ Name: "K04 full description", SomeProperty: "something" });
    }}
  >
    add more of K04
  </button>
```

How to get _SomeProperty_ in the output of step 2?
(hint: I am still on RHF v6, update to v7 planned, but I assume this is version independent)

**Reply by bluebill1049** [reply]
because there is no input as `SomeProperty`, and hence the value is not been included. however, the next feature release should solve this problem. https://github.com/react-hook-form/react-hook-form/pull/5069 which we will start to register for those absent inputs.

**Reply by wondering639** [reply]
> because there is no input as `SomeProperty`, and hence the value is not been included. however, the next feature release should solve this problem. #5069 which we will start to register for those absent inputs.

great, thanks :-) Will use a hidden field as temporary workaround.

**Reply by Aaronius** [reply]
I just came across this too. In our case, we toggle the rendering of certain fields based on the value of other fields. The field structure is complex enough that rendering hidden fields for all the values we want to retain isn't likely going to be feasible.

**Reply by bluebill1049** [reply]
From 7.6 onwards missing input from defaultValues will be registered and hidden inputs can be skipped.
